File: day_1/solution2.go
package main

import (
	_ "embed"
	"fmt"
	"strconv"
	"strings"
)

//go:embed input
var input string

func main() {
	var prev int64
	count := -1
	lines := strings.Split(input, "\n")

	for ndx := range lines {
		if ndx < 2 {
			continue
		}

		l1, _ := strconv.ParseInt(lines[ndx-2], 10, 32)
		l2, _ := strconv.ParseInt(lines[ndx-1], 10, 32)
		l3, _ := strconv.ParseInt(lines[ndx], 10, 32)
		curr := l1 + l2 + l3

		if curr > prev {
			count += 1
		}

		prev = curr
	}

	fmt.Println(count)
}

File: day_1/solution2.rb
# frozen_string_literal: true

require 'pry'
require 'pry-byebug'

prev = nil
count = 0
lines = IO.readlines('input')
lines.each_cons(3) do |l|
  curr = l[0].to_i + l[1].to_i + l[2].to_i
  count += 1 if not prev.nil? and curr > prev
  prev = curr
end

puts count
